### PR TITLE
add winflexbison 2.5.21

### PR DESCRIPTION
**Pull request by intelligide**
Specify library name and version:  **winflexbison/2.5.21**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/winflexbison/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+    def package_id(self):
+        self.info.include_build_settings()
+        del self.info.settings.arch
+        del self.info.settings.compiler
+        del self.info.settings.build_type
+
+    def package_info(self):
+        bindir = os.path.join(self.package_folder, "bin")
+        self.env_info.PATH.append(bindir)
```
Comment (Croydon) [inline recipes/winflexbison/all/conanfile.py]:
```suggestion
        self.output.info("Appending PATH environment variable: {}".format(bindir))
        self.env_info.PATH.append(bindir)
```

File: recipes/winflexbison/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
+import os
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+class WinflexbisonConan(ConanFile):
+    name = "winflexbison"
+    description = "Flex and Bison for Windows"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/uclouvain/openjpeg"
```
Comment (Croydon) [inline recipes/winflexbison/all/conanfile.py]:
```suggestion
    homepage = "https://github.com/lexxmark/winflexbison"
```

**Comment by Croydon** [comment]
You re-invented https://github.com/bincrafters/conan-winflexbison 🙃

**Comment by Croydon** [comment]
Feel free to copy over the Bincrafters recipe (which should work as it is)

File: recipes/winflexbison/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        return cmake
+
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy(pattern="data/*", dst="bin", src="{}/bison".format(self._source_subfolder), keep_path=True)
+        self.copy(pattern="FlexLexer.h", dst="include", src=os.path.join(self._source_subfolder, "flex", "src"), keep_path=False)
+        self.copy(pattern="COPYING", dst="licenses", src=os.path.join(self._source_subfolder, "bison", "src"), keep_path=False)
```
Comment (Croydon) [inline recipes/winflexbison/all/conanfile.py]: This doesn't contain all licenses

It lacks the license of flex `flex/src/COPYING`
and it lacks (at least) the license of `bison/data/skeletons` which we extracted from the a source file like this: https://github.com/bincrafters/conan-winflexbison/blob/43e3d812c71c4e0639fb373c51b9aef36414f48e/conanfile.py#L30

File: recipes/winflexbison/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        cmake.configure()
+        return cmake
+
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.build()
```
Comment (uilianries) [inline recipes/winflexbison/all/conanfile.py]:
```suggestion
        cmake.build()
        
    def _extract_license(self):
        with open(os.path.join(self._source_subfolder, "bison", "data", "skeletons", "glr.cc")) as f:
            content_lines = f.readlines()
        license_content = []
        for i in range(2, 16):
            license_content.append(content_lines[i][2:-1])
        tools.save("COPYING.GPL3", "\n".join(license_content))
```

File: recipes/winflexbison/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+    def package(self):
```
Comment (uilianries) [inline recipes/winflexbison/all/conanfile.py]:
```suggestion
    def package(self):
        self.copy(pattern="COPYING.GPL3", dst="licenses")
```

**Pull request closed without merging**